### ISSUE TITLE
react-server-cli does not work with npm 2.x

### ISSUE THREAD

**Issue by aickin** [redfin/react-server#34]
As reported in #2 at comment https://github.com/redfin/react-server/pull/2#issuecomment-188311662

To repro:
1. Use npm 2.x (I used 2.14.20)
1. `git clone https://github.com/doug-wade/example-reactserver.git`
1. `cd example-reactserver/`
1. `npm install`
1. `./node_modules/react-server-cli/bin/react-server-cli`

Gives this result:

```
2016-02-24T15:40:09.989Z - warning: [react-server-cli/index.js] PRODUCTION WARNING: the following current settings are discouraged in production environments. (If you are developing, carry on!):
2016-02-24T15:40:09.991Z - warning: [react-server-cli/index.js] -- Hot reload is enabled. Pass --hot=false, pass --production, or set NODE_ENV=production to turn off.
2016-02-24T15:40:09.991Z - warning: [react-server-cli/index.js] -- Minification is disabled. Pass --minify, pass --production, or set NODE_ENV=production to turn on.
2016-02-24T15:40:09.991Z - warning: [react-server-cli/index.js] -- NODE_ENV is not set to "production".
2016-02-24T15:40:13.584Z - notice: [react-server-cli/startServer.js] Starting servers...
2016-02-24T15:40:13.584Z - info: [react-server-cli/startServer.js] Starting hot reload JavaScript server...
2016-02-24T15:40:13.628Z - info: [react-server-cli/startServer.js] Starting HTML server...
2016-02-24T15:40:14.047Z - info: [react-server-cli/startServer.js] Started HTML server on port 3000
2016-02-24T15:40:14.174Z - info: [react-server-cli/startServer.js] Started hot reload JavaScript server on port 3001
2016-02-24T15:40:14.174Z - notice: [react-server-cli/startServer.js] Ready for requests on port 3000.
Hash: 22a0d3906fbc6214957a
Version: webpack 1.12.14
Time: 488ms
               Asset     Size  Chunks       Chunk Names
HelloWorld.bundle.js  22.3 kB       0       HelloWorld
chunk    {0} HelloWorld.bundle.js (HelloWorld) 64 bytes [rendered]
    [0] multi HelloWorld 64 bytes {0} [built] [4 errors]

ERROR in multi HelloWorld
Module not found: Error: Cannot resolve module 'webpack-dev-server/client' in /Users/sashaaickin/code/example-reactserver
 @ multi HelloWorld

ERROR in multi HelloWorld
Module not found: Error: Cannot resolve module 'webpack/hot/only-dev-server' in /Users/sashaaickin/code/example-reactserver
 @ multi HelloWorld

ERROR in multi HelloWorld
Module not found: Error: Cannot resolve module 'react-hot' in /Users/sashaaickin/code/example-reactserver
 @ multi HelloWorld

ERROR in multi HelloWorld
Module not found: Error: Cannot resolve module 'react-hot' in /Users/sashaaickin/code/example-reactserver
 @ multi HelloWorld
2016-02-24T15:40:14.974Z - debug: [react-server.core.renderMiddleware] Incoming request for /sockjs/info
2016-02-24T15:40:14.974Z - ok: [react-server.core.renderMiddleware] requestLocalStorageNamespaces val=6
2016-02-24T15:40:14.976Z - debug: [react-server.core.context.Navigator] Navigating to /sockjs/info?cb=7hgeqnbisy
```

**Comment by aickin** [comment]
Update: llooks like I can fix production mode if I install the following:

```
npm install --save babel-loader
npm install --save babel-runtime
```

Hot reload still doesn't work, though. I added `webpack-dev-server` and `react-hot-loader`, and it ended compilation errors, but there are still runtime errors on the client.

**Comment by aickin** [comment]
Fixed in 81b9802. Thanks!
